core/src/main/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProvider.scala

```diff
@@ -179,38 +182,44 @@ class LocalEntityProvider(workspace: Workspace, implicit protected val dataSourc
       operation <- update.operations
     } yield operation.name
 
-    withAttributeNamespaceCheck(namesToCheck) {
-      dataSource.inTransactionWithAttrTempTable (Set(AttributeTempTableType.Entity)){ dataAccess =>
-        val updateTrialsAction = dataAccess.entityQuery.getActiveEntities(workspaceContext, entityUpdates.map(eu => AttributeEntityReference(eu.entityType, eu.name))) map { entities =>
-          val entitiesByName = entities.map(e => (e.entityType, e.name) -> e).toMap
-          entityUpdates.map { entityUpdate =>
-            entityUpdate -> (entitiesByName.get((entityUpdate.entityType, entityUpdate.name)) match {
-              case Some(e) =>
-                Try(applyOperationsToEntity(e, entityUpdate.operations))
-              case None =>
-                if (upsert) {
-                  Try(applyOperationsToEntity(Entity(entityUpdate.name, entityUpdate.entityType, Map.empty), entityUpdate.operations))
-                } else {
-                  Failure(new RuntimeException("Entity does not exist"))
-                }
-            })
+    trace("LocalEntityProvider.batchUpdateEntitiesImpl") { rootSpan =>
+      rootSpan.putAttribute("workspaceId", OpenCensusAttributeValue.stringAttributeValue(workspaceContext.workspaceId))
+      rootSpan.putAttribute("isUpsert", OpenCensusAttributeValue.booleanAttributeValue(upsert))
+      rootSpan.putAttribute("entityUpdatesCount", OpenCensusAttributeValue.longAttributeValue(entityUpdates.length))
```

Comment:
I think it would be helpful here to get the number of operations as well as the number of entities. Is this overkill?

```suggestion
      rootSpan.putAttribute("entityUpdatesCount", OpenCensusAttributeValue.longAttributeValue(entityUpdates.length))
      rootSpan.putAttribute("entityOperationsCount", OpenCensusAttributeValue.longAttributeValue(entityUpdates.map(_.operations.length).sum))
```